### ISSUE TITLE
Pattern showing up on the texture for the controller model, controller input seems disabled

### ISSUE THREAD

**Issue by DaveyDangerpants** [ViveSoftware/ViveInputUtility-Unity#136]
We're getting a strange issue with the Vive controllers, which seems to come and go. We've tried a lot of different fixes, but nothing seems to work beyond restarting SteamVR a few times and then relaunching our program.

https://drive.google.com/file/d/1Vi8pz4rYZZmGNQCspgJWuv0sJiIJ6Abu/view?usp=sharing

The link shows a desktop capture of the issue in action. Sometimes, when we launch our program, one or both Vive controllers show a test pattern on the texture for the model. When this happens, the controller is unresponsive to input. Quitting the program, quitting SteamVR and relaunching SteamVR, turning on the Vive trackers and controllers and then relaunching the program often fixes it, but it can come and go unexpectedly.

Any idea what is happening here?

**Comment by chengnay** [comment]
@DaveyDangerpants What is the SteamVR Plugin version you are using?

**Comment by lupustom** [comment]
@DaveyDangerpants have you found a way to keep this from happening? I can second this issue ... relaunching SteamVR, the application or even the computer did not solve it.
@chengnay we are running the latest version of SteamVR (1.6.10) and did not experience the issue beforehand

**Comment by DaveyDangerpants** [comment]
@chengnay We're using SteamVR 1.5.2, but we've had the same issue with earlier versions of SteamVR as well.

@lupustom The issue arises intermittedly and we have tried many different solutions including:

- Using different versions of Unity
- Using different versions of the VIU plugin
- Using different versions of SteamVR
- Creating custom control bindings
- Using the default control bindings
- Pairing the trackers and controllers in different orders (in SteamVR)

Sometimes it seems like we've solved it, only to have it come up again later. It's very frustrating.

**Comment by DaveyDangerpants** [comment]
@chengnay I'll see if I can replicate it and attach the debug log.

**Comment by DaveyDangerpants** [comment]
[https://drive.google.com/file/d/1pcZwa_H7_5zicCU41PRq5meXAu79gw2T/view?usp=sharing](url)

A link to the debug log for this session:
[https://drive.google.com/file/d/16WBtnfqpGQZgRIu1sqwbrpIWyCncykH9/view?usp=sharing](url)

**Comment by chengnay** [comment]
@DaveyDangerpants Could you double check when the issue occurs, then you try to run SteamVR sample scene will also see the same issue?

**Comment by grrrwaaa** [comment]
The problem comes and goes. It just happened again, after I changed the bindings. Here is a clip from it happening in our project and in the VIU example scene. When I launch the Steam VR scene, the problematic controller disappears completely.

https://drive.google.com/file/d/1BBBz2_0TLHYttB0g-br_2Y4CoYT8mOos/view?usp=sharing

**Comment by chengnay** [comment]
@grrrwaaa In your video, did you release that one of your controllers is missing? I think is the one(left controller) that is having the issue.

**Comment by DaveyDangerpants** [comment]
> The problem comes and goes. It just happened again, after I changed the bindings. Here is a clip from it happening in our project and in the VIU example scene. When I launch the Steam VR scene, the problematic controller disappears completely.
> 
> https://drive.google.com/file/d/1BBBz2_0TLHYttB0g-br_2Y4CoYT8mOos/view?usp=sharing

@chengnay This was actually me. I was using a shared computer in a lab and didn't realize that I was logged into someone else's Git account.

The issue only happens when adding the VIU plugin into the project. We never experience this issue in any other project. Closing SteamVR and relaunching does not always fix the issue. As you can see in the quoted video, launching the VIU sample scene shows that the left controller does not work. Any ideas as to what is going on here?

**Comment by chengnay** [comment]
But from the video you provided, left controller was actually missing when playing SteamVR sample scene. Could you provide separate logs, failed log in scene that uses VIU plugin and failed log in SteamVR sample scene?

**Comment by DaveyDangerpants** [comment]
Yes, it appears that in our project as well ast he VIU sample scene, the left controller displayed that strange pattern and was unresponsive to input. When the SteamVR sample scene was launched, it disappeared completed (but was still showing up as paired and active in the SteamVR application).

When it happens again, I'll send along logs for both the VIU sample scene and SteamVR sample scene.

**Comment by chengnay** [comment]
@DaveyDangerpants Thanks, if you can provide error logs that will be helpful.
I will try to replicate this issue myself to see what is actually causing the problem.
In VIU, we actually getting controller's component states from SteamVR, if SteamVR is not responsive, that is why those strange patterns show on the controller model.

> but was still showing up as paired and active in the SteamVR application

Do you mean in the scene with VIU plugin it is not paired and inactive? If not paired and inactive, how can your controller shows pose? I don't see why disappeared controller can consider as active?

**Comment by lupustom** [comment]
@DaveyDangerpants do we know if the issue persists in actual builds or only in the editor?

**Comment by DaveyDangerpants** [comment]
> Do you mean in the scene with VIU plugin it is not paired and inactive? If not paired and inactive, how can your controller shows pose? I don't see why disappeared controller can consider as active?

@chengnay That is what is so strange about this bug. SteamVR is showing the controller as paired and active, but it shows up with that strange pattern in our project as well as the VIU example scene, and disappears completely in the SteamVR sample scene. It is behaving as if it isn't there, but it clearly is, as demonstrated by the appearance of the controller in our project and the VIU example scene. 

> do we know if the issue persists in actual builds or only in the editor?

The issue does persists in actual builds. Sometimes, what happens is that multiple controllers appear on top of each other. As a workaround, for now, we are simply using the controllers' transform for interaction in our project, eschewing any button interaction via the controller altogether. Clearly, this is not an ideal workaround, but at least the positional tracking seems to work, even if the buttons do not.

**Comment by DaveyDangerpants** [comment]
Here is an error log from the error occurring in all three scenes (our project, the VIU example scene and the SteamVR scene) 

https://drive.google.com/file/d/19SMU_1WdtM26Zp0d-mka9I05vqgk6dqQ/view?usp=sharing

**Comment by DaveyDangerpants** [comment]
And another one:
https://drive.google.com/file/d/17KFUqeqRA7XFQwAjWy4nw6JgVj_P0sGs/view?usp=sharing

**Comment by chengnay** [comment]
@DaveyDangerpants I found an closed issue thread in SteamVR Plugin GitHub. I posted your log for reference.
